app/controllers/api/v1/workflows_controller.rb

```diff
@@ -79,23 +90,17 @@ def field_context
     end
   end
 
-  # This is a very good reason to move away from the api controller work we've got
-  # it's hard to hook into the right place to run actions that in theory should
-  # be easily located in specific controller actions
+  # Allways update the state of the workflow & selector services
+  # when modifying the subject sets or retired_subjects links
   def post_link_actions(workflow)
-    if workflow.set_member_subjects.exists?
+    relation_index = %w(subject_sets retired_subjects).index(relation.to_s)
```

Comment:
It might be me, but this reads very strangely to me. I would much prefer just a standard `case` statement, even if that then has a duplicated line for NotifySelectorOfChange in both of the cases. By the time I get to the `.zero?` (or `== 0`) we've lost all meaning of what that zero is.

Reply:
updated - i was trying to be too cute.